x/deployment/types/events_test.go

```diff
@@ -0,0 +1,174 @@
+package types
+
+import (
+	"fmt"
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	"github.com/pkg/errors"
+
+	"github.com/ovrclk/akash/sdkutil"
+)
+
+var (
+	keyAcc = sdk.NewKVStoreKey(auth.StoreKey)
+	//keyParams = sdk.NewKVStoreKey(params.StoreKey)
+
+	errWildcard = errors.New("wildcard string error can't be matched...")
+)
+
+type testEventParsing struct {
+	msg    sdkutil.Event
+	expErr error
+}
+
+func (tep testEventParsing) testMessageType() func(t *testing.T) {
+	ev, err := ParseEvent(tep.msg)
+	return func(t *testing.T) {
+		if tep.expErr != nil && err != tep.expErr {
+			if tep.expErr != errWildcard {
+				t.Errorf("unexpected error: %v exp: %v", err, tep.expErr)
+				t.Logf("%T %v", errors.Cause(err), err)
+			} else {
+				t.Logf("expected wildcard error: %v", err)
+			}
+		}
+		if ev == nil {
+			t.Logf("ParseEvent returned value is nil, TODO: return parsed data")
+		}
+	}
+}
+
+var TEPS = []testEventParsing{
+	{
+		msg: sdkutil.Event{
+			Type: "nil",
+		},
+		expErr: sdkutil.ErrUnknownType,
+	},
+	{
+		msg: sdkutil.Event{
+			Type: sdkutil.EventTypeMessage,
+		},
+		expErr: nil,
+	},
+
+	{
+		msg: sdkutil.Event{
+			Type:   sdkutil.EventTypeMessage,
+			Module: ModuleName,
+		},
+		expErr: nil,
```

Comment:
Looking at [this](https://github.com/ovrclk/akash/blob/e126a9d1838a8b369221188e97b5b895239d87fe/x/deployment/types/event.go#L116-L139), I believe these things should cause an error:

* `Action` is unpopulated
* `Action` is neither `evActionDeploymentCreate` nor `evActionDeploymentUpdate`
* `DeploymentID` attributes are not set.

Reply:
Yeah... I don't know how I missed that, and clearly the err assertion isn't working.

Reply:
I think **LAMBDA BOWL** is causing one of those annoying closure-in-a-loop bugs.